packages/block-library/src/cover/style.scss

```diff
@@ -302,3 +292,32 @@ section.wp-block-cover-image > h2,
 		color: inherit;
 	}
 }
+
+// Backward compatibility for v14 deprecation.
+.wp-block-cover:not(.has-no-zindex-in-inner-blocks) {
+	.wp-block-cover__video-background {
+		z-index: z-index(".wp-block-cover__video-background");
+	}
+
+	.wp-block-cover__image-background {
+		z-index: z-index(".wp-block-cover__image-background");
+	}
+
+	// The first selector is required for old Cover markup/
+	// Keep .wp-block-cover__gradient-background for v8 deprecation.
+	&.has-background-dim:not(.has-background-gradient)::before,
+	.wp-block-cover__background,
+	.wp-block-cover__gradient-background {
+		z-index: z-index(".wp-block-cover.has-background-dim::before");
+	}
+
+	.wp-block-cover__inner-container {
+		z-index: z-index(".wp-block-cover__inner-container");
+
+		// Reset the z-index to auto when the body has a modal open. So when the
+		// modal is inside the cover, it doesn't create a stacking context.
+		@at-root .has-modal-open & {
+			z-index: auto;
+		}
```

Comment:
I think it's good to have this anyway, so it's automatically fixed in all the sites just by updating the Gutenberg. Make sense?

Reply:
Maybe my memory is failing me but the original issue was in the editor only correct? The interactivity API was used on the frontend and that was all displaying ok.

In the editor, with a deprecation in place, the blocks would be migrated and thus fixed without needing the superfluous styles. So the problem would still be automatically fixed by updating Gutenberg only without the unnecessary styles and the block modifying document level classes.

Reply:
The main issue was on the frontend (we had a few reports about that from users). And as a lower impact, it was also happening in the editor.

Reply:
Thanks for setting me straight on that. 

So, we can leave this style there but the [`has-modal-open` class being added](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation/edit/index.js?plain=1#L87) to the document wrapper should be removed. The deprecation should sort out the migration and display of the block in the editor so we no longer need the change that was introduced in https://github.com/WordPress/gutenberg/pull/66093.

Avoiding that was the main driver for this whole approach in my view.

Reply:
I thought the same, but I decided to keep given this reason:

> As part of this PR, I also thought of removing [this effect that adds the has-modal-open in the editor](https://github.com/WordPress/gutenberg/pull/66093/files#diff-884dae78f1676d454c049107e9b05b2403eb042457f77f836bd170d104396b8bR89-R97) because that wouldn't be needed for this purpose anymore. But thinking more, I kept it to keep the consistency there between the frontend and the editor.

If you reconsider, and think it's still worth removing it, I can remove it. But I think it would be better to keep this for consistency (frontend - editor) for the future.

> Avoiding that was the main driver for this whole approach in my view.

I think the benefit is that we only use it for BC and not as our main solution anymore.

Reply:
> If you reconsider, and think it's still worth removing it, I can remove it. But I think it would be better to keep this for consistency (frontend - editor) for the future.

To me, a block mutating the classes on the overall document is very odd in the editor. It also attracted similar reactions from other contributors on that original PR.

I appreciate the consistency argument and I would normally lean that way myself. In this scenario though, without a genuine use case or purpose, it would be best not to keep that. It could also encourage improper coupling of blocks, other components etc. to that class. 

If the situation changes in the future we could always revisit bringing in that consistency. That might even be via the interactivity API itself and be truly consistent with the frontend then. This approach would also then allow a dedicated discussion around that behaviour without a rush to a decision.

TL;DR - Let's remove the addition of the `has-modal-open` class for now.

Reply:
Gotcha! Reverted here: https://github.com/WordPress/gutenberg/pull/66249/commits/7226f4bd33ed00ca3c2effa254c8c31e7aedd2a7

I'll probably continue the rest tomorrow. And thanks for the commits you added here. 😉

Reply:
@andrewserong , I think you are referring to the `@at-root .has-modal-open &`, right?

If yes, I think it's good to keep it for backward compatibility. With this, it gets fixed in the frontend for the users without needing to re-save the post.